### ISSUE TITLE
UnnecessaryParentheses warns about xor-like operations

### ISSUE THREAD

**Issue by imanushin** [detekt/detekt#2789]
## Expected Behavior
No detekt warnings is appeared.

## Observed Behavior
The warning is appeared - UnnecessaryParentheses - [XXX] at /Users/zzz/XXX.kt:21:17

## Steps to Reproduce
Code:
```kotlin
data class(val id1: Int?, val id2: Int?) {
   init {
      require((id1 === null && id2 !== null) || (id1 !== null && id2 === null)) {
         "..."       
      }
   }
}
```

## Context
Brackets above are needed for readability, to simulate xor-like behaviour (when one and only one field can have value)

## Your Environment
* Version of detekt used: 1.9.1
* Version of Gradle used (if applicable): 6.5.0
* Operating System and version: n/a
* Link to your project (if it's a public repository): private repo

**Comment by schalkms** [comment]
We won’t invest more time in this rule, since detekt complies with IntelliJ’s rule.
That’s why I’ll close this.

Reference: https://github.com/detekt/detekt/blob/1e712757bb399666e26514740b08e2faa565bf7b/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParentheses.kt#L50

**Comment by schalkms** [comment]
If you prefer readability here, please add a Suppress annotation. We discussed this rule in other issues. It’s just not worth the time to apply more exceptions to this rule. It makes the rule unnecessarily complex.
